scripts/globals/porter_moogle_util.lua

```diff
@@ -0,0 +1,297 @@
+    local slipId, slipCount = getSlipId(trade);
+    if (slipId == 0 or slipCount > 1) then return; end;
+	
+    local storableItemIds = getStorableItems(player, trade, slipId);
+	
+    if (#storableItemIds > 0) then
+        storeItems(player, storableItemIds, slipId, e);
+    else
+        startRetrieveProcess(player, e.RETRIEVE_EVENT_ID, slipId);
+    end 
+end
+
+----------------------------------------------------------------------
+-- desc : Retrieves the selected item from storage, removes it from
+--        the slip's extra data, displays a message to the user, and
+--        updates the user's event data.
+----------------------------------------------------------------------
+function porterEventUpdate(player, csid, option, RETRIEVE_EVENT_ID, RETRIEVE_DIALOG_ID, ITEM_CANNOT_BE_OBTAINED)
+    local slipId = player:getLocalVar('slipId');
+    if (csid == RETRIEVE_EVENT_ID and slipId ~= 0 and slipId ~= nil) then
+        local extra = player:getRetrievableItemsForSlip(slipId);
+        local itemsOnSlip = getItemsOnSlip(extra, slipId);
+        local retrievedItemId = itemsOnSlip[option + 1];
+        
+        if (player:hasItem(retrievedItemId) or player:getFreeSlotsCount() == 0) then
+            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED, retrievedItemId);
+        else
+            local k = find(slipItems[slipId], retrievedItemId);
+            local extraId = math.floor((k - 1) / 8);
+            local bitmask = extra[extraId + 1];
+            if bitmask < 0 then
+                bitmask = bitmask + 256;
+            end
+            bitmask = bit.band(bitmask, bit.bnot(bit.lshift(1, (k - 1) % 8)));
+            extra[extraId + 1] = bitmask;
+
+            player:retrieveItemFromSlip(slipId, retrievedItemId, extraId, bitmask);
+            player:messageSpecial(RETRIEVE_DIALOG_ID, retrievedItemId, nil, nil, retrievedItemId, false);
+        end
+        
+        local params = int8ToInt32(extra);
+        player:updateEvent(params[1], params[2], params[3], params[4], params[5], params[6], slipId);
+    end 
+end
+
+----------------------------------------------------------------------
+-- desc : Completes the event.
+----------------------------------------------------------------------
+function porterEventFinish(player, csid, option, TALK_EVENT_ID, ITEM_CANNOT_BE_OBTAINED, ITEM_OBTAINED, NOT_HAVE_ENOUGH_GIL)
+    if (csid == TALK_EVENT_ID and option < 1000) then
+        -- This is just because hilarious.
+        option = math.floor(option / 16) + (option % 16);
+        local hasItem = player:hasItem(slipIds[option]);
+        if (hasItem or player:getFreeSlotsCount() == 0) then
+            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED, slipIds[option]);
+            return;
+        end
+
+        if (player:getGil() < 1000) then
+            player:messageSpecial(NOT_HAVE_ENOUGH_GIL, slipIds[option]);
+            return;
+        end
+        
+        if (player:delGil(1000)) then
```

Comment:
combine into one check (so player:delGil instead of getGil) cause i get to be picky as well
as far as i could tell, they do like exactly the same shit

if it doesn't work, i will take full responsibility and blame lautan